Assignment.Tests/PingProcessTests.cs

```diff
@@ -0,0 +1,179 @@
+    //{
+    //    (int exitCode, string? stdOutput) = Sut.Run("badaddress");
+    //    Assert.IsFalse(string.IsNullOrWhiteSpace(stdOutput));
+    //    stdOutput = WildcardPattern.NormalizeLineEndings(stdOutput!.Trim());
+    //    Assert.AreEqual<string?>(
+    //        "Ping request could not find host badaddress. Please check the name and try again.".Trim(),
+    //        stdOutput,
+    //        $"Output is unexpected: {stdOutput}");
+    //    Assert.AreEqual<int>(1, exitCode);
+    //}
+
+    //[TestMethod]
+    //public void Run_CaptureStdOutput_Success()
+    //{
+    //    PingResult result = Sut.Run("localhost");
+    //    AssertValidPingOutput(result);
+    //}
+
+    //[TestMethod]
+    //public void RunTaskAsync_Success()
+    //{
+    //    // Do NOT use async/await in this test.
+    //    // Test Sut.RunTaskAsync("localhost");
+    //    var pingProcess = new PingProcess();
+    //    var result = pingProcess.RunTaskAsync("localhost").Result;
+
+    //    Assert.IsNotNull(result);
+    //}
+
+    //[TestMethod]
+    //public void RunAsync_UsingTaskReturn_Success()
+    //{
+    //    var pingProcess = new PingProcess();
+    //    var result = pingProcess.RunAsync("localhost").Result;
+
+    //    Assert.IsNotNull(result);
+    //}
+
+    //[TestMethod]
+
+    //async public Task RunAsync_UsingTpl_Success()
+    //{
+    //    var pingProcess = new PingProcess();
+    //    var result = await pingProcess.RunAsync("localhost");
+
+    //    Assert.IsNotNull(result);
+    //}
+
+
+
+    //[TestMethod]
+    //[ExpectedException(typeof(AggregateException))]
+    //public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrapping()
+    //{
+    //    var pingProcess = new PingProcess();
+    //    var cancellationTokenSource = new CancellationTokenSource();
+    //    cancellationTokenSource.CancelAfter(1000); // Cancel after 1 second
+    //    Assert.ThrowsException<AggregateException>(() =>
+    //    {
+    //        var result = pingProcess.RunAsync("localhost", cancellationTokenSource.Token).Result;
+    //    });
+    //}
+
+    //[TestMethod]
+    //[ExpectedException(typeof(TaskCanceledException))]
+    //public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrappingTaskCanceledException()
+    //{
+    //    var pingProcess = new PingProcess();
+    //    var cancellationTokenSource = new CancellationTokenSource();
+    //    cancellationTokenSource.CancelAfter(1000); // Cancel after 1 second
+    //    Assert.ThrowsException<TaskCanceledException>(() =>
+    //    {
+    //        var result = pingProcess.RunAsync("localhost", cancellationTokenSource.Token).Result;
+    //    });
+    //    // Use exception.Flatten()
+    //}
+
+    //[TestMethod]
+    //async public Task RunAsync_MultipleHostAddresses_True()
+    //{
+    //    // Pseudo Code - don't trust it!!!
+    //    string[] hostNames = new string[] { "localhost", "localhost", "localhost", "localhost" };

```

Comment:
Can make this simpler
```suggestion
    //    string[] hostNames = ["localhost", "localhost", "localhost", "localhost"];

```

Reply:
thanks